From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 1 Sep 2026 21:01:14 +0200
Subject: fix(busybox): do not install losetup applet

The busybox `losetup` applet does not support the option `--show`.

Fixes: https://github.com/dracut-ng/dracut/issues/2675
Part of: https://github.com/dracut-ng/dracut/issues/2437

Origin: upstream, https://github.com/dracut-ng/dracut/commit/cd772fb43a2303adac5714ae53f9d4174db2d4ac
---
 modules.d/10busybox/module-setup.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules.d/10busybox/module-setup.sh b/modules.d/10busybox/module-setup.sh
index f175f72..8d5e208 100755
--- a/modules.d/10busybox/module-setup.sh
+++ b/modules.d/10busybox/module-setup.sh
@@ -28,6 +28,11 @@ install() {
             # See https://github.com/vda-linux/busybox_mirror/issues/33
             continue
         fi
+        if [[ ${_path##*/} == losetup ]]; then
+            # The busybox losetup applet does not support the option --show.
+            # See https://github.com/vda-linux/busybox_mirror/issues/37
+            continue
+        fi
         if [[ ${_path##*/} == mke2fs ]]; then
             # e2fsprogs provides mke2fs. mkfs.ext2, mkfs.ext3, and mkfs.ext4 are symlinks to it.
             # Busybox does not provide a mkfs.ext4 applet and mkfs.ext* would point to busybox then.
